libsession/src/main/java/org/session/libsession/messaging/sending_receiving/pollers/Poller.kt

```diff
@@ -152,14 +152,18 @@ class Poller(private val configFactory: ConfigFactoryProtocol, debounceTimer: Ti
                     Log.w("Loki", "shared config message handled in configs wasn't SharedConfigurationMessage but was ${message.javaClass.simpleName}")
                     return@forEach
                 }
-                forConfigObject.merge(hash!! to message.data)
-                latestMessageTimestamp = if ((message.sentTimestamp ?: 0L) > (latestMessageTimestamp ?: 0L)) { message.sentTimestamp } else { latestMessageTimestamp }
+                val merged = forConfigObject.merge(hash!! to message.data).firstOrNull { it == hash }
```

Comment:
I might have to look at these again when the libsession library merges this into dev, I might also have to check the latestMessageTimestamp is being called, probably remove the `SnodeAPI.nowWithOffset` in that case

Reply:
This code can be condensed a lot.

Does `forConfigObject#merge` have side-effects?

If not, it can be condensed even further.

Reply:
I guess the only side effects would be internal in the `forConfigObject`